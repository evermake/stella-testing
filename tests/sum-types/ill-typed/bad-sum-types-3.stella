language core;
extend with #sum-types;

fn g(x : Nat + (Bool + (fn(Nat) -> Nat))) -> Nat {
  return match x {
      inl(n) => succ(n)
    | inr(bf) => match bf {
          inr(f) => f(f(succ(0)))
        | inl(b) => b
      }
  }
}

fn main(x : Nat) -> Nat {
  return g(inr(inr(fn(n : Nat) { return g(inl(n)) })))
}
