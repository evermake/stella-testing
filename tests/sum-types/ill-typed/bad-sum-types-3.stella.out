An error occurred during typechecking!
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
expected type
  Nat
but got
  Bool
for expression
  b